masonry_core/src/passes/layout.rs

```diff
@@ -191,6 +191,29 @@ pub(crate) fn run_layout_on<W: Widget + ?Sized>(
     new_size
 }
 
+// --- MARK: CLEAR LAYOUT
+// This function is called on stashed widgets and their children
+// to set all layout flags to false.
+fn clear_layout_flags(
+    mut widget: ArenaMut<'_, Box<dyn Widget>>,
+    state: ArenaMut<'_, WidgetState>,
+    properties: ArenaMut<'_, AnyMap>,
+) {
+    state.item.needs_layout = false;
+    state.item.request_layout = false;
+
+    let id = state.item.id;
+    recurse_on_children(
+        id,
+        widget.reborrow_mut(),
+        state.children,
+        properties.children,
+        |widget, state, properties| {
+            clear_layout_flags(widget, state, properties);
```

Comment:
Would it be correct to `debug_assert` that `state.item.is_stashed` is true here?